core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -795,32 +799,63 @@ private[zio] final class FiberContext[E, A](
     case _                   =>
   }
 
-  @tailrec
-  private[this] final def kill0(
-    k: Callback[Nothing, Exit[E, A]]
-  ): Option[IO[Nothing, Exit[E, A]]] = {
+  private[this] final def kill0(fiberId: FiberId): UIO[Exit[E, A]] = {
+    @tailrec
+    def setInterruptedLoop(): Unit = {
+      val oldState = state.get
+
+      oldState match {
+        case Executing(FiberStatus.Suspended(true, _), observers, interrupted) =>
+          if (!state.compareAndSet(
+                oldState,
+                Executing(FiberStatus.Running, observers, interrupted ++ Cause.interrupt(fiberId))
```

Comment:
The state is getting switched to `FiberStatus.Running` here - just verifying that is correct

Reply:
Yes, we set it to `FiberStatus.Running` because we immediately call `evaluateNow` which means the fiber will be running again.